src/system/arduino/esp32/system.c

```diff
@@ -117,7 +117,13 @@ int8_t _z_condvar_wait(_z_condvar_t *cv, _z_mutex_t *m) { return pthread_cond_wa
 /*------------------ Sleep ------------------*/
 int z_sleep_us(unsigned int time) { return usleep(time); }
 
-int z_sleep_ms(unsigned int time) { return usleep(time * 1000U); }
+int z_sleep_ms(unsigned int time) {
+    for (unsigned int i = 0; i < time; i++) {
+        z_sleep_us(1000);
```

Comment:
Most sleep APIs promise to sleep _at least_ whatever you asked them to. This may compound, so this approach may make sleeps longer than expected:
Arguably, we might just decide to accept this compounding if we measure it to be small enough for our application.

Reply:
Nice catch! You are right.
I will include an extra check on how much it slept and continue subsequent sleeps accordingly, so we can minimize the amount of extra time.
The important point is that we still sleep at least `time`.

Reply:
Yeah, current implementation will oversleep by x‰, where x is the average oversleep in µs of `z_sleep_us`. In micros, I would expect this number could be surprisingly high, so a deadline might be a good thing to add

Reply:
Indeed, this was accumulating more than I expected:
  - Sleeping 10000 ms OP (i.e., `for` loop): 13.071s 12.625s 12.623s
  - Sleeping 10000 ms now: 10.363s 10.006s 10.005s